silverback/types.py

```diff
@@ -1,9 +1,20 @@
+from enum import Enum
 from typing import Optional, Protocol
 
 from pydantic import BaseModel
 from typing_extensions import Self  # Introduced 3.11
 
 
+class TaskType(str, Enum):
```

Comment:
question: does having `str` here do anything if you still have to override `__str__` ?

Reply:
Makes it do `isinstance(task_type, str)`

You also get `__str__` method but it contains more than just value

Reply:
In Python 3.11 there is a true `StrEnum` that does this for you

```py
In [2]: class E(str, Enum):
   ...:     A = "a"
   ...:     B = "b"
   ...: 

In [3]: str(E.A)
Out[3]: 'E.A'

In [4]: class E(StrEnum):
   ...:     A = "a"
   ...:     B = "b"
   ...: 

In [5]: str(E.A)
Out[5]: 'a'
```

Reply:
[5ad9b7a](https://github.com/ApeWorX/silverback/pull/66/commits/5ad9b7a43ba33e967f87773052b1e2599c78f02c)